crates/trie/parallel/src/root.rs

```diff
@@ -237,6 +241,7 @@ impl From<ParallelStateRootError> for ProviderError {
             ParallelStateRootError::StorageRoot(StorageRootError::Database(error)) => {
                 Self::Database(error)
             }
+            ParallelStateRootError::Other(other) => Self::Database(DatabaseError::Other(other)),
```

Comment:
this should not be casted to database error. out of scope for this PR, but we have to fix this at some point ^^